src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ChainedComparator.java

```diff
@@ -0,0 +1,27 @@
+package io.stargate.sgv2.jsonapi.service.operation.model;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.FindOperation;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.ReadDocument;
+import io.stargate.sgv2.jsonapi.util.JsonNodeComparator;
+import java.util.Comparator;
+import java.util.List;
+
+public record ChainedComparator(List<FindOperation.OrderBy> sortColumns)
+    implements Comparator<ReadDocument> {
+  @Override
+  public int compare(ReadDocument o1, ReadDocument o2) {
+    int i = 0;
+    final List<JsonNode> sortValuesO1 = o1.sortColumns();
```

Comment:
is this what ReadOperation would create for each document right?

Reply:
Yes.